pkg/api/webassets/webassets.go

```diff
@@ -133,24 +133,35 @@ func readWebAssets(r io.Reader) (*dtos.EntryPointAssets, error) {
 	// 	return nil, fmt.Errorf("missing swagger entry")
 	// }
 
-	rsp := &dtos.EntryPointAssets{
-		JSFiles: make([]dtos.EntryPointAsset, 0, len(entryPoints.App.Assets.JS)),
-		Dark:    entryPoints.Dark.Assets.CSS[0],
-		Light:   entryPoints.Light.Assets.CSS[0],
-		Swagger: make([]dtos.EntryPointAsset, 0, len(entryPoints.Swagger.Assets.JS)),
+	rsp := new(dtos.EntryPointAssets)
+
+	if entryPoints.Dark != nil && len(entryPoints.Dark.Assets.CSS) > 0 {
+		rsp.Dark = entryPoints.Dark.Assets.CSS[0]
 	}
 
-	for _, entry := range entryPoints.App.Assets.JS {
-		rsp.JSFiles = append(rsp.JSFiles, dtos.EntryPointAsset{
-			FilePath:  entry,
-			Integrity: integrity[entry],
-		})
+	if entryPoints.Light != nil && len(entryPoints.Light.Assets.CSS) > 0 {
+		rsp.Light = entryPoints.Light.Assets.CSS[0]
 	}
-	for _, entry := range entryPoints.Swagger.Assets.JS {
-		rsp.Swagger = append(rsp.Swagger, dtos.EntryPointAsset{
-			FilePath:  entry,
-			Integrity: integrity[entry],
-		})
+
+	if entryPoints.App != nil {
```

Comment:
should any of these error if they are nil?

Reply:
if anything is nil the frontend will not work.  It will only be nil if "yarn build" (or whatever variant you use) has not been run.

Perhaps a better approach is a check above all this to says:

```
if entryPoints.App == nil || entryPoints.Light == nil || entryPoints.Darkt == nil || entryPoints.Swaggert == nil {
   return fmt.Errorf("missing frontend assets.  check that yarn build has run")
}   
```
or similar

Reply:
Fixed!